torch/autograd/_functions/tensor.py

```diff
@@ -601,13 +609,93 @@ def forward(self, input):
         return torch.cumsum(input, dim=self.dim)
 
     def backward(self, grad_output):
-        grad_input = torch.cumsum(-grad_output, dim=self.dim)
+        return reverse_cumsum(grad_output, dim=self.dim)
+        
+class Cumprod(Function):
+    
+    def __init__(self, dim):
```

Comment:
This is an old style autograd function, @apaszke do we absolutely want this to be a new style class or is it ok?